.github/workflows/pytest.yml

```diff
@@ -31,16 +30,23 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v3
+      # We check out only a limited depth and then pull tags to save time
+      - name: Checkout source
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 100
```

Comment:
Why 100 here?  I think the default is just 1, so I'd have thought that this would tend to be slower, not faster.  Although either way this step is pretty fast relative to the pytest step, so maybe not worth optimizing anyway?

Reply:
I think we need to fetch some commits to get the proper version using versionner with versionner.py when using `pip install --no-deps .` in further steps.

Example in this workflow :

`Successfully installed pvlib-0.9.1+17.g3fcda76`

Without this you cannot infer version 0.9.1 with 17 commit ahead

100 is purely arbitrary